TST, MAINT: test_quadpack TestCtypesQuad wasn't fully migrated to pytest

There's a rarely-used codepath that depends on `unittest` module, and can fail on some machines with more unusual setups, as I noticed today:

`python runtests.py --mode=full -t "scipy/integrate/tests/test_quadpack.py"`

```
========================================================================================================================================================== ERRORS ==========================================================================================================================================================
______________________________________________________________________________________________________________________________________ ERROR at setup of TestCtypesQuad.test_typical _______________________________________________________________________________________________________________________________________
scipy/integrate/tests/test_quadpack.py:51: in setup_method
    self.skipTest("Ctypes can't import libm.so")
E   AttributeError: 'TestCtypesQuad' object has no attribute 'skipTest'
____________________________________________________________________________________________________________________________________ ERROR at setup of TestCtypesQuad.test_ctypes_sine _____________________________________________________________________________________________________________________________________
scipy/integrate/tests/test_quadpack.py:51: in setup_method
    self.skipTest("Ctypes can't import libm.so")
E   AttributeError: 'TestCtypesQuad' object has no attribute 'skipTest'
__________________________________________________________________________________________________________________________________ ERROR at setup of TestCtypesQuad.test_ctypes_variants ___________________________________________________________________________________________________________________________________
scipy/integrate/tests/test_quadpack.py:51: in setup_method
    self.skipTest("Ctypes can't import libm.so")
E   AttributeError: 'TestCtypesQuad' object has no attribute 'skipTest'
=============================================================================================================================================== 29 passed, 3 errors in 2.36s ===============================================================================================================================================
```

This should be an easy fix with use of an imperative `pytest.skip()` instead now that the test class simply does not inherit from standard lib `TestCase` or whatever. I'll open a PR momentarily, I think.

--- a/scipy/integrate/tests/test_quadpack.py
+++ b/scipy/integrate/tests/test_quadpack.py
@@ -4,6 +4,7 @@ import numpy as np
 from numpy import sqrt, cos, sin, arctan, exp, log, pi, Inf
 from numpy.testing import (assert_,
         assert_allclose, assert_array_less, assert_almost_equal)
+import pytest
 from pytest import raises as assert_raises
 
 from scipy.integrate import quad, dblquad, tplquad, nquad
@@ -46,7 +47,7 @@ class TestCtypesQuad(object):
         else:
             # This test doesn't work on some Linux platforms (Fedora for
             # example) that put an ld script in libm.so - see gh-5370
-            self.skipTest("Ctypes can't import libm.so")
+            pytest.skip("Ctypes can't import libm.so")
 
         restype = ctypes.c_double
         argtypes = (ctypes.c_double,)

--- a/scipy/integrate/tests/test_quadpack.py
+++ b/scipy/integrate/tests/test_quadpack.py
@@ -5,7 +5,6 @@ from numpy import sqrt, cos, sin, arctan, exp, log, pi, Inf
 from numpy.testing import (assert_,
         assert_allclose, assert_array_less, assert_almost_equal)
 import pytest
-from pytest import raises as assert_raises
 
 from scipy.integrate import quad, dblquad, tplquad, nquad
 from scipy._lib._ccallback import LowLevelCallable
@@ -89,7 +88,7 @@ class TestCtypesQuad(object):
         for j, func in enumerate(all_sigs):
             callback = LowLevelCallable(func)
             if func in legacy_only_sigs:
-                assert_raises(ValueError, quad, callback, 0, pi)
+                pytest.raises(ValueError, quad, callback, 0, pi)
             else:
                 assert_allclose(quad(callback, 0, pi)[0], 2.0)
 
@@ -98,7 +97,7 @@ class TestCtypesQuad(object):
             if func in legacy_sigs:
                 assert_allclose(quad(func, 0, pi)[0], 2.0)
             else:
-                assert_raises(ValueError, quad, func, 0, pi)
+                pytest.raises(ValueError, quad, func, 0, pi)
 
 
 class TestMultivariateCtypesQuad(object):

